packages/fxa-settings/src/models/reliers/base-relier.ts

```diff
@@ -73,6 +86,12 @@ export class BaseRelier implements ModelContextProvider, Relier {
     return 'base';
   }
 
+  /** Lazy loaded client info. */
+  clientInfo: Promise<RelierClientInfo> | undefined;
```

Comment:
Question I had here in Slack was around what's all making this async... `serviceName` in particular is used quite a lot so it would be ideal to do something like `const { serviceName } = useRelier();` but we can otherwise maybe we can (not in this PR) create a custom hook that runs that `useEffect` to grab it.

Reply:
I've filed a follow up for this. I think it's possible the client info doesn't need to fetch with and xhr at all. I think we could just be written to the page directly, but currently make a call using the OAuth client to fetch this info.